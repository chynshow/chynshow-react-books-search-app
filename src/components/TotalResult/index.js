import React, { useContext } from 'react';
import { AppContext } from '../../state/AppContext';

const TotalResult = () => {
  // Get data from global app state
  const { totalResults, loading } = useContext(AppContext);
  return (
    <>
      <div>
        {!loading && totalResults !== null && (
          <span>
            We found{' '}
            {totalResults > 1
              ? `${totalResults} results for you!`
              : `${totalResults} result for you!`}
          </span>
        )}

        {totalResults === null && (
          <span>You don't have any books yet! Start search!</span>
        )}
      </div>
    </>
  );
};

export default TotalResult;
