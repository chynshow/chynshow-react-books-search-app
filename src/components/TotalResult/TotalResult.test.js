import React from 'react';
import TotalResult from './';
import { shallow } from 'enzyme';

it('should renders TotalResult component', () => {
  const c = shallow(<TotalResult />);
  expect(c).toMatchSnapshot();
});
