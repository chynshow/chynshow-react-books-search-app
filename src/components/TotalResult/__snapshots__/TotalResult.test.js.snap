// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders TotalResult component 1`] = `
<Fragment>
  <div
    className="c-total-result"
  >
    <span
      className="c-total-result__text"
    >
      You don't have any results yet!
    </span>
  </div>
</Fragment>
`;
